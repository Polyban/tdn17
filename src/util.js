/* speech building helper function */
export function wrapIn (tag) {
  return (utterance) => `<${tag}>${utterance}</${tag}>`
}

export function pitchDown (pitch='-10%') {
  return (utterance) => `<prosody pitch="${pitch}">${utterance}</prosody>`
}

export function getCanonicalSlotResolution (slot) {
  if (!slot.resolutions ||Â !slot.resolutions.resolutionsPerAuthority
    || !slot.resolutions.resolutionsPerAuthority.length) {
    return null
  }
  return slot.resolutions.resolutionsPerAuthority[0].values[0].value
}