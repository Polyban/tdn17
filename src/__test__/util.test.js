// import interactionModel from '../interaction-model.json'
import { getCanonicalSlotResolution } from '../util'

describe('util', () => {
  it('should getCanonicalSlotResolution', () => {
    const mockSlot = {
      name: 'Tag',
      value: 'freitag',
      resolutions: {
        resolutionsPerAuthority: [
          { values: [{ value: { id: 'FRIDAY' } }] }
        ]
      }
    }
    const slot = getCanonicalSlotResolution(mockSlot)
    expect(slot).toMatchSnapshot()
    expect(getCanonicalSlotResolution({})).toMatchSnapshot()
  })
})
