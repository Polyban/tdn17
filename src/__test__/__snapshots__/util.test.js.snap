// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util should getCanonicalSlotResolution 1`] = `
Object {
  "id": "FRIDAY",
}
`;

exports[`util should getCanonicalSlotResolution 2`] = `null`;
